app/controllers/jobseekers/registrations_controller.rb

```diff
@@ -1,38 +1,52 @@
 class Jobseekers::RegistrationsController < Devise::RegistrationsController
+  before_action :set_update_password, only: %i[edit update]
   before_action :check_password_difference, only: %i[update]
   before_action :check_new_password_presence, only: %i[update]
+  before_action :check_email_difference, only: %i[update]
   after_action :replace_devise_notice_flash_with_success!, only: %i[destroy update]
   after_action :remove_devise_notice_flash!, only: %i[create]
 
-  def check_your_email; end
-
 protected
 
   def check_password_difference
+    return unless @update_password
     return unless params[resource_name][:password].present?
     return unless params[resource_name][:current_password] == params[resource_name][:password]
 
     render_resource_with_error(:password, :same_as_old)
   end
 
   def check_new_password_presence
+    return unless @update_password
     return unless params[resource_name][:current_password].present?
     return if params[resource_name][:password].present?
 
     render_resource_with_error(:password, :too_short)
   end
 
+  def check_email_difference
+    return if @update_password
+    return unless params[resource_name][:email].present?
+    return unless params[resource_name][:email] == current_jobseeker.email
+
+    render_resource_with_error(:email, :same_as_old)
+  end
+
   def render_resource_with_error(field, error)
     self.resource = resource_class.new
     resource.errors.add(field, error)
     render :edit
   end
 
+  def set_update_password
+    @update_password = params[:update_password] == "true" || params[:commit] == t("buttons.update_password")
```

Comment:
Does rubocop object to:
```ruby
    @update_password = (params[:update_password] == "true") || (params[:commit] == t("buttons.update_password"))
```

Bit easier on the old eyes imo.

Reply:
I prefer minimal code, and operators precedence is something that should be well known

Reply:
It's more that I have to re-parse the sentence a few times as I go along. https://en.wikipedia.org/wiki/Garden-path_sentence#Examples

I'll leave it to Connor's taste.